### ISSUE TITLE
Transactions display as 0.00

### ISSUE THREAD

**Issue by fswings** [moneymanagerex/android-money-manager-ex#344]
Most transactions which imported in from CSV via desktop app (1.12 and 1.20) are displayed as 0.00.

<img src="https://cloud.githubusercontent.com/assets/5247221/7997008/ca32da0e-0b1e-11e5-9a45-5e46019d91d6.png" width="200px" />

The transactions which **are not affected** are:

1. Transfers to/from other accounts.
2. Manual input transactions on desktop app.
3. Manual input transactions on Android app.

All of the transactions which are displayed as "0.00" have in common that  they were imported into desktop app (tried with both 1.12 and 1.20) from csv files.

**Comment by MisterY** [comment]
Hi! Is this a bug in Android app or the desktop app? In case it relates to the desktop version, the desktop version has a separate issue tracker.
Currently I see no CSV import or export functionality in the Android app.

**Comment by fswings** [comment]
Hi, the screenshot is from the android app, I believe the issue relates to the android app as all of the transactions are visible correctly on the desktop app. 

I've got LG G3 running stock Lollipop. 

I've worked out that transactions that the Android app displays as 0.00 are the ones which I imported into the desktop app from CSV. 

However, transfers and direct entries in the desktop app display fine in the android app. Also transactions created on the Android app display fine on the mobile app. 

The database is shared between and desktop and mobile via Dropbox. 

I hope this helps.

**Comment by MisterY** [comment]
Thanks. I'll have to try that specific scenario. It seems weird at the first glance as the amount field should be the same in all these cases.

**Comment by fswings** [comment]
For me it smells of a database related issue. Wouldn't surprise me if was encoding related but I haven't e tried changing the CSV encoding yet. 

The CSV file was generated by exporting from Microsoft Excel 2010 on Windows. Desktop app was running on Windoes 7 as well.

**Comment by MisterY** [comment]
Would you mind peeking directly into the database using something like http://sqlitebrowser.org/ ?
This should provide instant feedback on the differences between the valid and invalid records. 
In case you are correct about the encoding, that would then probably not even show up in my test scenario.

**Comment by fswings** [comment]
I'll give it a go when I get home later.

**Comment by fswings** [comment]
OK, I think I found the problem. The Column TRANSAMOUNT in the CHECKINGACCOUNT_V1 has all of the correct values however the Column TOTRANSAMOUNT had zero for the data imported from CSV. 

It seems the desktop app shows the TRANSAMOUNT column data whereas in the android app you show the TOTRANSAMOUNT only. 

So it could be the desktop app isn't filling the TOTRANSAMOUNT column properly or it knows if TOTRANSAMOUNT is zero then use the TRANSAMOUNT column.

**Comment by fswings** [comment]
![capture](https://cloud.githubusercontent.com/assets/5247221/8041111/dc3f98f8-0e0d-11e5-9b9a-4f0544746f4b.png)

The left hand column is TRANSAMOUNT and the right hand most column is TOTRANSAMOUNT.

Have you been able to reproduce the error?

What is the difference in behaviour between Android app and desktop?

**Comment by MisterY** [comment]
Hi, I am away until Wednesday, the 17th, without access to a PC. Need to see what the difference is. My guess is that the import should populate the same fields as when a transaction is entered by hand, but need to confirm with the desktop team.

**Comment by MisterY** [comment]
I have created the issue #501 in the desktop app repository to see which component should be corrected.

**Comment by vomikan** [comment]
Don't use "totransamount". It's only for transfers.
08 июня 2015 г. 20:50 пользователь "Alen Siljak" <notifications@github.com>
написал:

**Comment by MisterY** [comment]
I see. It still looks strange that all the transactions are using this field except the imported ones.
The reason we switched the query to use the To Amount was the fact that qif only uses the destination amount. And all the other transactions had the correct value in this field.
We can create some logic that checks the the type of transaction and the amount and this will slow down the data fetching significantly. CSV import could be modified to populate this field and create the transactions that are aligned with the others.

**Comment by fswings** [comment]
I did a test by entering a transaction in both the mobile app and desktop app. Both transactions (not transfers) add data to both of these columns. 
Importing CSVs to the desktop app only fills in one of the columns. 
I've also encountered a strange issue on the desktop app version 1.2.x vs 1.12, where the most recent version no longer imports in CSV files which were successfully imported in version 1.12.
My instincts are telling me that perhaps these issues are related. 
I'm trying to create a sample CSV file to share  to help expose this behaviour that doesn't include my actual data. Hand created CSV files pose no problems but it seeks to be those exported from Excel appear to be troublesome. Will do some more testing and feedback. 
--- Constantly Mobile ---
On 11 Jun 2015 12:10, Alen Siljak <notifications@github.com> wrote:I see. It still looks strange that all the transactions are using this field except the imported ones.
The reason we switched the query to use the To Amount was the fact that qif only uses the destination amount. And all the other transactions had the correct value in this field.
We can create some logic that checks the the type of transaction and the amount and this will slow down the data fetching significantly. CSV import could be modified to populate this field and create the transactions that are aligned with the others.
